redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/cache/RedisCache.java

```diff
@@ -223,7 +223,8 @@ public <T> CompletableFuture<T> get(Object key, Argument<T> requiredType, Suppli
                     Optional<T> deserialized = valueSerializer.deserialize(data, requiredType);
                     boolean hasValue = deserialized.isPresent();
                     if (expireAfterAccess != null && hasValue) {
-                        return redisKeyAsyncCommands.expire(serializedKey, expireAfterAccess).thenApply(ignore -> deserialized.get());
+                        final long expireAfterAccessInSeconds = expireAfterAccess / 1000;
+                        return redisKeyAsyncCommands.expire(serializedKey, expireAfterAccessInSeconds).thenApply(ignore -> deserialized.get());
```

Comment:
Instead of converting to milliseconds (which could lose precision), maybe just use `pexpire`? (https://redis.io/docs/latest/commands/pexpire/)

Reply:
indeed , using pexpire would solve the issue rather than converting to milliseconds, I've added tests that hit the methods .